File: python/car_plant.py
from mitsubishi_triton import MitsubishiTriton
from toyota_corolla import ToyotaCorolla
from toyota_corolla_sports import ToyotaCorollaSports
from sand_buggy import SandBuggy
from sand_buggy_lite import SandBuggyLite

class CarPlant:

    log = []

    @staticmethod
    def makeToyotaCorolla():
        return ToyotaCorolla(CarPlant.log)

    @staticmethod
    def makeToyotaCorollaSports():
        return ToyotaCorollaSports(CarPlant.log)

    @staticmethod
    def makeMitsubishiTriton():
        return MitsubishiTriton(CarPlant.log)

    @staticmethod
    def makeSandBuggy():
        return SandBuggy(CarPlant.log)

    @staticmethod
    def makeSandBuggyLite():
        return SandBuggyLite(CarPlant.log)

File: java/src/com/xpsurgery/carplant/CarPlant.java
package com.xpsurgery.carplant;

import java.io.Writer;

public class CarPlant {

	public static Car makeToyotaCorolla(Writer listener) {
		return new ToyotaCorolla(listener);
	}

	public static Car makeToyotaCorollaSports(Writer listener) {
		return new ToyotaCorollaSports(listener);
	}

	public static Car makeMitsubishiTriton(Writer listener) {
		return new MitsubishiTriton(listener);
	}

	public static Car makeSandBuggy(Writer listener) {
		return new SandBuggy(listener);
	}

	public static Car makeSandBuggyLite(Writer listener) {
		return new SandBuggyLite(listener);
	}
}

